src/utils/handleSubmitDirty.ts

```diff
@@ -51,8 +51,23 @@ const formDataFilterForDirtyHelper = (dirtyFields: any, formData: any) => {
   }
 
   if (typeof dirtyFields === "object") {
+    const extendedDiryFields = JSON.parse(JSON.stringify(dirtyFields))
+
+    Object.entries(dirtyFields).forEach(([key]) => {
+      const keyIsDirtyFrom = DIRTY_KEYS_TO_KEEP.find(
+        (keeyToKeep) => keeyToKeep === key
+      )
+      const allfields = formData[keyIsDirtyFrom]
+
+      if (allfields)
+        Object.entries(allfields).forEach(([fieldKey, value]) => {
+          if (!extendedDiryFields[keyIsDirtyFrom].key)
```

Comment:
What is `.key` in this condition refer to?

Reply:
for example: 'eventSources' is the the 'keyIsDirtyFrom'  and 'eventBrite' is the .'key'

Reply:
Hmm, doesn't this solution cause things like `socialLinks` to be always sent?

Reply:
No, just in that case when some social links are dirty.

Reply:
Hmm I don't get why this condition is necessary, it is always true for form values that don't have a `key` field?

I'm thinking if the problem can be solved by modifying the `DIRTY_KEYS_TO_KEEP` logic a little:
```ts
DIRTY_KEYS_TO_KEEP.forEach((keyToKeep) => {
  if (dirtyFields[keyToKeep]) {
    newObj[keyToKeep] = formData[keyToKeep]
  }
})
```
and placing these lines right above the `return !isEmpty ? newObj : TO_FILTER_FLAG` seems to be working:
```ts
    const isEmpty = Object.keys(newObj).length <= 0

    // Re-attach id field to dirty objects
    if (!isEmpty && "id" in formData) {
      newObj.id = formData.id
    }
```
It also seems to be working fine for the case when I remove the last social link